### DISCUSSION TITLE
HW1部署至Azure時失敗

### DISCUSSION THREAD

**Discussion by handsomeniko** [1101-GS3073-C/Python-Community-GS3073-C#54] [Q&A] [answered]
今天嘗試把寫好的HW1部署到Azure時遇到了一些狀況
我自己用VScode跑是能正常使用的，如下圖
![login](https://user-images.githubusercontent.com/48316857/136789868-77073e54-c502-4172-ac08-cbf5543fd7c2.png)

下面簡單說明我的步驟:

我先把檔案全部放到我自己的GitHub網頁上，如下圖
![git](https://user-images.githubusercontent.com/48316857/136790882-94926272-9ff6-45d5-a129-d41c174ccb1d.png)

因為原本放在1101-GS3073-C的資料庫是Private，在Azure上設定時無法連接到
所以我自己開了一個Public的資料庫放我的檔案，不知道是不是GitHub的資料庫要設定什麼

接下來我按照老師的影片去Azure部署我的網站，資料如下
![azure](https://user-images.githubusercontent.com/48316857/136789384-46625145-f0a1-41c9-8493-76e191cbf1ec.png)

![app2](https://user-images.githubusercontent.com/48316857/136791900-7defcd44-ef18-4dc7-be94-041e98b01083.png)

而且GitHub中的action也顯示成功
![git_action](https://user-images.githubusercontent.com/48316857/136789879-2d292b8d-0993-44e2-8c4e-cf73474c1429.png)

但最後打開部屬好的網站
[https://nikoflask.azurewebsites.net/](url)

卻顯示錯誤
![app](https://user-images.githubusercontent.com/48316857/136790042-1ebddd80-93cc-4c12-a631-85d875892c0a.png)

不知道有沒有大神知道我哪裡做錯了QQ

**Comment by chia-kai-chang** [comment, marked as answer]
app.py裡面有`import pandas as pd`
但是requirements.txt裡面沒有pandas

**Reply by handsomeniko** [reply]
@benctw 謝謝老師!!剛剛改了之後成功了~

**Reply by chia-kai-chang** [reply]
Tag User後，就會有App push notifications 
![image](https://user-images.githubusercontent.com/1768891/136912213-b243d29d-d41c-42c3-ab5a-fa0b70c31c19.png)
